crates/blockchain/mempool_tx_pruner.rs

```diff
@@ -0,0 +1,105 @@
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use crate::{Blockchain, error::MempoolError};
+use ethrex_common::types::MempoolTransaction;
+use ethrex_storage::error::StoreError;
+use spawned_concurrency::{
+    messages::Unused,
+    tasks::{CastResponse, GenServer, GenServerHandle, send_interval},
+};
+use tracing::{debug, info, warn};
+
+// Amount of seconds after which we prune old entries from mempool (We should fine tune this)
+const PRUNE_WAIT_TIME_SECS: u128 = 300; // 5 minutes
+
+// Amount of seconds between each prune
+const PRUNE_INTERVAL_SECS: u64 = 300; // 5 minutes
+
+// Number of microseconds in a second
+const MICROS_PER_SECOND: u128 = 1_000_000;
+
+#[derive(Debug, Clone)]
+pub struct MempoolPruner {
+    blockchain: Arc<Blockchain>,
+}
+
+#[derive(Debug, Clone)]
+pub enum InMessage {
+    Prune,
+}
+
+#[derive(Debug, Clone)]
+pub enum OutMessage {
+    Done,
+}
+
+impl MempoolPruner {
+    pub async fn spawn(
+        blockchain: Arc<Blockchain>,
+    ) -> Result<GenServerHandle<MempoolPruner>, MempoolPrunerError> {
+        info!("Starting Mempool Pruner");
+
+        let state = MempoolPruner { blockchain };
+
+        let server = state.clone().start();
+
+        send_interval(
+            Duration::from_secs(PRUNE_INTERVAL_SECS),
+            server.clone(),
+            InMessage::Prune,
+        );
+
+        Ok(server)
+    }
+
+    pub fn prune(&self) -> Result<(), MempoolPrunerError> {
+        let now = SystemTime::now().duration_since(UNIX_EPOCH)?.as_micros();
+        let filter_fn = |tx: &MempoolTransaction| {
+            now.saturating_sub(tx.time()) > PRUNE_WAIT_TIME_SECS * MICROS_PER_SECOND
```

Comment:
I don't know if this is the logic that we want long term. What does the spec say regarding pruning transactions? How do other clients behave?

Also maybe we should investigate a bit which properties long running mempool txs have:
- To low of a base fee?
- Nonce to high?
- Another?

I suspect that if we have 200k txs then there might be something that we're doing wrongly, perhaps not cleaning them properly after they are included in a block for example.